okhttp/src/jvmMain/kotlin/okhttp3/internal/-UtilJvm.kt

```diff
@@ -322,3 +325,12 @@ internal inline fun Any.assertThreadDoesntHoldLock() {
     throw AssertionError("Thread ${Thread.currentThread().name} MUST NOT hold lock on $this")
   }
 }
+
+internal fun threadPoolExecutor(
+  corePoolSize: Int = 0,
+  maximumPoolSize: Int = Int.MAX_VALUE,
+  keepAliveTime: Long = 60,
+  timeUnit: TimeUnit = TimeUnit.SECONDS,
+  workingQueue: BlockingQueue<Runnable> = SynchronousQueue(),
+  threadFactory: ThreadFactory = threadFactory("$okHttpName Dispatcher", false)
```

Comment:
This feels wrong with Dispatcher in the thread name. Can you revert this part of the change and try to focus the PR on just new functionality?

Reply:
should I revert the constructor and `executorServiceOrNull` too? or just this function?

Reply:
Let's wait for @swankjesse, Dispatcher is different than our other types.  If we start adding new mutable fields, we should discuss where we are heading.

For now, maybe just move this to an internal method in Dispatcher.